sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2052,6 +2052,20 @@ object SQLConf {
       .stringConf
       .createWithDefault("")
 
+  val FASTFAIL_ON_FILEFORMAT_OUTPUT =
+    buildConf("spark.sql.execution.fastFailOnFileFormatOutput")
+      .internal()
+      .doc("Whether to fast fail task execution when writing output to FileFormat datasource. " +
+        "If this is enabled, in `FileFormatWriter` we will catch `FileAlreadyExistsException` " +
+        "and fast fail output task without further task retry. Only enabling this if you know " +
+        "the `FileAlreadyExistsException` of the output task is unrecoverable, i.e., further " +
+        "task attempts won't be able to success. If the `FileAlreadyExistsException` might be " +
+        "recoverable, you should keep this as disabled and let Spark to retry output tasks. " +
+        "This is disabled by default.")
+      .version("3.1.0")
```

Comment:
cc @zsxwing and @cloud-fan .
Do you guys want this conf in `master` only?

Reply:
I think it's better to merge to branch-3.0 as well as currently user don't have a flag for this in 3.0.x.

Reply:
Then, I guess `3.0.2` might be better here. Given that 3.1.0 is rescheduled, 3.0.2 will arrive before or at the same time because it's easier than 3.1.0 release.

Reply:
Ok. I changed it to `3.0.2`.